packages/esm-patient-orders-app/src/lab-results/lab-results.resource.ts

```diff
@@ -0,0 +1,165 @@
+  display: string;
+  uuid: string;
+  name: string;
+  locale: string;
+  localePreferred: boolean;
+  conceptNameType: string;
+}
+
+export interface Datatype {
+  uuid: string;
+  display: string;
+  name: string;
+  description: string;
+  hl7Abbreviation: string;
+  retired: boolean;
+  resourceVersion: string;
+}
+
+export interface Description {
+  display: string;
+  uuid: string;
+  description: string;
+  locale: string;
+  resourceVersion: string;
+}
+
+export interface Mapping {
+  display: string;
+  uuid: string;
+  conceptReferenceTerm: OpenmrsResource;
+  conceptMapType: OpenmrsResource;
+  resourceVersion: string;
+}
+
+export function useOrderConceptByUuid(uuid: string) {
+  const apiUrl = `${restBaseUrl}/concept/${uuid}?v=${labConceptRepresentation}`;
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<{ data: LabOrderConcept }, Error>(
+    apiUrl,
+    openmrsFetch,
+  );
+  return {
+    concept: data?.data,
+    isLoading,
+    isError: error,
+    isValidating,
+    mutate,
+  };
+}
+
+export function useLabEncounter(encounterUuid: string) {
+  const apiUrl = `${restBaseUrl}/encounter/${encounterUuid}?v=${labEncounterRepresentation}`;
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<FetchResponse<Encounter>, Error>(
+    apiUrl,
+    openmrsFetch,
+  );
+
+  return {
+    encounter: data?.data,
+    isLoading,
+    isError: error,
+    isValidating,
+    mutate,
+  };
+}
+
+export function fetchObservation(obsUuid: string) {
+  return openmrsFetch(`${restBaseUrl}/obs/${obsUuid}?v=${conceptObsRepresentation}`).then(({ data }) => {
+    if (data) {
+      return data;
+    }
+    return null;
+  });
+}
+
+const addObservation = async (encounterUuid: string, obsPayload: any, abortController: any) => {
+  const saveResultObs = await openmrsFetch(`${restBaseUrl}/encounter/${encounterUuid}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    signal: abortController.signal,
+    body: obsPayload,
+  });
+
+  return saveResultObs.status;
+};
+
+const editObservation = async (obsUuid: string, obsPayload: any, abortController: any) => {
+  const editResultObs = await openmrsFetch(`${restBaseUrl}/obs/${obsUuid}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    signal: abortController.signal,
+    body: obsPayload,
+  });
+
+  return editResultObs.status;
+};
+
+//TODO: the calls to update order and observations for results should be transactional to allow for rollback
+export async function updateOrderResult(
+  orderUuid: string,
+  encounterUuid: string,
+  obsUuid: string,
+  obsPayload: any,
+  fulfillerPayload: any,
+) {
+  const abortController = new AbortController();
+  const saveObs = obsUuid
+    ? editObservation(obsUuid, obsPayload, abortController)
+    : addObservation(encounterUuid, obsPayload, abortController);
+
+  saveObs.then((obsStatus) => {
+    if (obsStatus === 200) {
```

Comment:
What happens if the server responds with a 201 status code?

Reply:
The response gives a 200 code but i've added 201 as well to future proof it

Reply:
I'd probably have changed `editObservation()` and `addObservation()` to just return the full `FetchResponse` or return `response.ok` instead of the actual status code.